src/fluree/db/query/fql.cljc

```diff
@@ -50,9 +172,13 @@
    (if (cache? query-map)
      (cache-query ds query-map)
      (go-try
-       (let [pq (parse/parse-query query-map)
-             oq (<? (optimize/optimize ds pq))]
-         (<? (exec/query ds tracker oq)))))))
+       (if-let [spec (simple-count-distinct-subject-spec query-map)]
```

Comment:
My thought is this function could be greatly simplified by checking a parsed query instead of the raw query map. I know parsing the whole query takes more time, but I think most of the savings from this optimization come from avoiding the deserialization costs for all the flakes.

And since the vast majority of our queries are not simple count distinct subject queries, this extra parsing that checks for this special case slows down every other query.